File: ccif/src/java/other/org/himalay/ccif/HTMLCapabilities.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class HTMLCapabilities extends BinStruct implements PublicBinMsg {




    // members variables
    // linkSupport
    public short linkSupport ;
    // tableSupport
    public short tableSupport ;
    // listSupport
    public short listSupport ;
    // imageSupport
    public short imageSupport ;

    public HTMLCapabilities () // throws Exception
    {
        init();
    }

    private void init()
    {
        // Initialize linkSupport

        // Initialize tableSupport

        // Initialize listSupport

        // Initialize imageSupport

    }

    public int readNoHeader(DataInputStream istream) throws IOException
    {

        return read(istream);
    }

    public int read(DataInputStream istream) throws IOException
    {
        preRead();
        int retVal= 0;

        // read linkSupport
        {
            linkSupport=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read tableSupport
        {
            tableSupport=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read listSupport
        {
            listSupport=(short)(istream.readUnsignedByte());
            retVal+=1;
        }
        // read imageSupport
        {
            imageSupport=(short)(istream.readUnsignedByte());
            retVal+=1;
        }

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException
    {
        preWrite();
        int retVal= 0;






        // write linkSupport
        ostream.writeByte(linkSupport);
        retVal +=1;
        // write tableSupport
        ostream.writeByte(tableSupport);
        retVal +=1;
        // write listSupport
        ostream.writeByte(listSupport);
        retVal +=1;
        // write imageSupport
        ostream.writeByte(imageSupport);
        retVal +=1;
        postWrite();
        return retVal;
    }

    public int dump(DumpContext dc) throws IOException
    {
        dc.indent();
        dc.getPs().print("HTMLCapabilities\n");
        dc.increaseIndent();
        int retVal= 0;
        // write linkSupport
        dc.indent();
        dc.getPs().println("linkSupport="+linkSupport+"(0x"+ Integer.toHexString(linkSupport) + ")" );
        // write tableSupport
        dc.indent();
        dc.getPs().println("tableSupport="+tableSupport+"(0x"+ Integer.toHexString(tableSupport) + ")" );
        // write listSupport
        dc.indent();
        dc.getPs().println("listSupport="+listSupport+"(0x"+ Integer.toHexString(listSupport) + ")" );
        // write imageSupport
        dc.indent();
        dc.getPs().println("imageSupport="+imageSupport+"(0x"+ Integer.toHexString(imageSupport) + ")" );
        dc.decreaseIndent();
        return retVal;
    }


    // Getter for linkSupport
    //public short getLinkSupport()
    //{
    //    return linkSupport ;
    //}


    // Setter for linkSupport
    //public void setLinkSupport(short val)
    //{
    //    this.linkSupport= val;
    //}
    // Getter for tableSupport
    //public short getTableSupport()
    //{
    //    return tableSupport ;
    //}


    // Setter for tableSupport
    //public void setTableSupport(short val)
    //{
    //    this.tableSupport= val;
    //}
    // Getter for listSupport
    //public short getListSupport()
    //{
    //    return listSupport ;
    //}


    // Setter for listSupport
    //public void setListSupport(short val)
    //{
    //    this.listSupport= val;
    //}
    // Getter for imageSupport
    //public short getImageSupport()
    //{
    //    return imageSupport ;
    //}


    // Setter for imageSupport
    //public void setImageSupport(short val)
    //{
    //    this.imageSupport= val;
    //}


    public int getSize() throws IOException
    {
        DataOutputStream dos= new DataOutputStream(new NullStream());
        return this.write(dos);
    }



}

// End of code

File: libccif/src/bp/HTMLCapabilities.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:12 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "HTMLCapabilities.h"

#ifdef org_himalay_ccif_HTMLCapabilities__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
HTMLCapabilities::HTMLCapabilities() {
    references= 0;


    // linkSupport
    linkSupport= (ui8)0;
    // tableSupport
    tableSupport= (ui8)0;
    // listSupport
    listSupport= (ui8)0;
    // imageSupport
    imageSupport= (ui8)0;
}
// Destructor
HTMLCapabilities::~HTMLCapabilities() {
    // linkSupport

    // tableSupport

    // listSupport

    // imageSupport

}

int HTMLCapabilities::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // linkSupport
    {
        linkSupport=(istream.readUI8());
        retVal+=1;
    }
    // tableSupport
    {
        tableSupport=(istream.readUI8());
        retVal+=1;
    }
    // listSupport
    {
        listSupport=(istream.readUI8());
        retVal+=1;
    }
    // imageSupport
    {
        imageSupport=(istream.readUI8());
        retVal+=1;
    }
    return retVal;
}

int HTMLCapabilities::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read linkSupport
    {
        linkSupport=(istream.readUI8());
        retVal+=1;
    }
    // read tableSupport
    {
        tableSupport=(istream.readUI8());
        retVal+=1;
    }
    // read listSupport
    {
        listSupport=(istream.readUI8());
        retVal+=1;
    }
    // read imageSupport
    {
        imageSupport=(istream.readUI8());
        retVal+=1;
    }

    return retVal;
}

int HTMLCapabilities::write(DataOutputStream& ostream) { // throws IOException





    int retVal= 0;
    // write linkSupport
    ostream.writeUI8(linkSupport);
    retVal +=1;
    // write tableSupport
    ostream.writeUI8(tableSupport);
    retVal +=1;
    // write listSupport
    ostream.writeUI8(listSupport);
    retVal +=1;
    // write imageSupport
    ostream.writeUI8(imageSupport);
    retVal +=1;

    return retVal;
}

int HTMLCapabilities::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("HTMLCapabilities\n");
    dc.increaseIndent();
    int retVal= 0;
    // write linkSupport
    dc.indent();
    dc.getPs().print("linkSupport=");
    dc.getPs().println((long)linkSupport);
    // write tableSupport
    dc.indent();
    dc.getPs().print("tableSupport=");
    dc.getPs().println((long)tableSupport);
    // write listSupport
    dc.indent();
    dc.getPs().print("listSupport=");
    dc.getPs().println((long)listSupport);
    // write imageSupport
    dc.indent();
    dc.getPs().print("imageSupport=");
    dc.getPs().println((long)imageSupport);
    dc.decreaseIndent();
    return retVal;
}


// Getter for linkSupport
//ui8 HTMLCapabilities::getLinkSupport()
//{
//return this->linkSupport;
//}

// Setter for linkSupport
//void HTMLCapabilities::setLinkSupport(ui8 val)
//{
//this->linkSupport= val;
//}

// Getter for tableSupport
//ui8 HTMLCapabilities::getTableSupport()
//{
//return this->tableSupport;
//}

// Setter for tableSupport
//void HTMLCapabilities::setTableSupport(ui8 val)
//{
//this->tableSupport= val;
//}

// Getter for listSupport
//ui8 HTMLCapabilities::getListSupport()
//{
//return this->listSupport;
//}

// Setter for listSupport
//void HTMLCapabilities::setListSupport(ui8 val)
//{
//this->listSupport= val;
//}

// Getter for imageSupport
//ui8 HTMLCapabilities::getImageSupport()
//{
//return this->imageSupport;
//}

// Setter for imageSupport
//void HTMLCapabilities::setImageSupport(ui8 val)
//{
//this->imageSupport= val;
//}

int HTMLCapabilities::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_HTMLCapabilities__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::HTMLCapabilities* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::HTMLCapabilities* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code